src/user-event/setup/setup.ts

```diff
@@ -143,13 +145,33 @@ function createInstance(config: UserEventConfig): UserEventInstance {
 
   // We need to bind these functions, as they access the config through 'this.config'.
   const api = {
-    press: press.bind(instance),
-    longPress: longPress.bind(instance),
-    type: type.bind(instance),
-    clear: clear.bind(instance),
-    scrollTo: scrollTo.bind(instance),
+    press: wrapAndBindImpl(instance, press),
+    longPress: wrapAndBindImpl(instance, longPress),
+    type: wrapAndBindImpl(instance, type),
+    clear: wrapAndBindImpl(instance, clear),
+    scrollTo: wrapAndBindImpl(instance, scrollTo),
   };
 
   Object.assign(instance, api);
   return instance;
 }
+
+// This implementation is sourced from testing-library/user-event
+// https://github.com/testing-library/user-event/blob/7a305dee9ab833d6f338d567fc2e862b4838b76a/src/setup/setup.ts#L121
+function wrapAndBindImpl<Impl extends (...args: any) => Promise<any>>(
```

Comment:
Seems like we could move it to utils next to `asyncWrapper`

Reply:
Actually I moved the `asyncWrapper` cause it's used also by `waitFor` so I don't think it should be in the `userEvent` folder. I don't see `wrapAndBindImpl` being used elsewhere as it's directly tied to user event so I think it's in the right place